hermes/src/state/cache.rs

```diff
@@ -206,17 +208,40 @@ impl AggregateCache for crate::state::State {
             let key = message_state.key();
             let time = message_state.time();
             let cache = message_cache.entry(key).or_insert_with(BTreeMap::new);
-
             cache.insert(time, message_state);
 
             // Remove the earliest message states if the cache size is exceeded
             while cache.len() > self.cache.cache_size as usize {
                 cache.pop_first();
             }
         }
+
         Ok(())
     }
 
+    /// This method takes the current feed ids and prunes the cache for the keys
+    /// that are not present in the current feed ids.
+    ///
+    /// There is a side-effect of this: if a key gets removed, we will
+    /// lose the cache for that key and cannot retrieve it for historical
+    /// price queries.
+    async fn prune_removed_keys(&self, current_keys: HashSet<MessageStateKey>) {
+        let mut message_cache = self.cache.message_cache.write().await;
+
+        // Sometiems, some keys are removed from the accumulator. We track which keys are not
+        // present in the message states and remove them from the cache.
+        let keys_in_cache = message_cache
+            .iter()
+            .map(|(key, _)| key.clone())
+            .collect::<HashSet<_>>();
+
+        for key in keys_in_cache {
+            if !current_keys.contains(&key) {
+                message_cache.remove(&key);
```

Comment:
Since this is not very often, it would be great if we add some log here to see it in action once live.

Reply:
I added a info log level for it.